File: JAVA/MatMult.java
import java.util.Scanner;
public class MatMult
{
	public static void main(String []args)
	{
		int i,j,k;
		Scanner sc=new Scanner(System.in);
		System.out.println("Enter the dimentions of Matrix A:");
		int m1=sc.nextInt();
		int n1=sc.nextInt();
		System.out.println("Enter the dimention of Matrix B:");
		int m2=sc.nextInt();
		int n2=sc.nextInt();
		
		int a[][]=new int[m1][n1];
		int b[][]=new int[m2][n2];
		int c[][]=new int[m2][n1];
		
		
		if(n1==m2)
		{
			System.out.println("Enter the Elements of Matrix A:");
			for(i=0;i<m1;i++)
			{
				for(j=0;j<n1;j++)
				{
					a[i][j]=sc.nextInt();
				}
			}
			
			System.out.println("Enter the Elements of Matrix B:");
			for(i=0;i<m2;i++)
			{
				for(j=0;j<n2;j++)
				{
					b[i][j]=sc.nextInt();
				}
			}
			
			System.out.println("Elements of Matrix A:");
			for(i=0;i<m1;i++)
			{
				for(j=0;j<n1;j++)
				{
					System.out.print(" "+a[i][j]);
				}
				System.out.println("");
			}
			
			System.out.println("Elements of Matrix B:");
			for(i=0;i<m2;i++)
			{
				for(j=0;j<n2;j++)
				{
					System.out.print(" "+b[i][j]);
				}
				System.out.println("");
			}
			 
	
			for(i=0;i<m1;i++)
			{
				for(j=0;j<n2;j++)
				{
					c[i][j]=0;
					{
					   for(k=0;k<m2;k++)
					  {
						c[i][j]+=a[i][k]*b[k][j];
					  }
					}
				}
			}
			
			System.out.println("Multiplication of Matrix A and B is:");
			{
				for(i=0;i<m1;i++)
			{
				for(j=0;j<n2;j++)
				{
					System.out.print("   "+c[i][j]);
				}
				System.out.println("");
			}
			}			
		}
		else
		{
			System.out.println("Multiplication not possible");
		}
	}
}

File: C/maTmult.c
#include<stdio.h>

void accept(int x[10][10],int m1,int n1)
{
    int i,j;
    printf("Enter array elements:");
    for(i=0;i<m1;i++)
    {
        for(j=0;j<n1;j++)
        {
            scanf("%d",&x[i][j]);
        }
    }
}

void mult(int a[10][10],int b[10][10],int c[10][10],int m1, int n1,int m2, int n2)
{
    int i,j,k;
    for(i=0;i<m1;i++)
    {
        for(j=0;j<n2;j++)
        {
            c[i][j]=0;
            {
                for(k=0;k<n1;k++)
                {
                    c[i][j]+=a[i][j]*b[i][j];
                }
            }
        }
    }
}


void display(int c[10][10],int m1,int n2)
{
    int i,j;
    printf("Multiplication=\n");
    for(i=0;i<m1;i++)
    {
        for(j=0;j<n2;j++)
        {
            printf("%3d",c[i][j]);
        }
        printf("\n");
    }
}
int main()
{
    int a[10][10],b[10][10],c[10][10],m1,n1,m2,n2;
    printf("Enter  degree of matrix 1:");
    scanf("%d%d",&m1,&n1);
    printf("\n");
    printf("Enter degree of matrix 2:");
    scanf("%d%d",&m2,&n2);
    printf("\n");
    {
        if(n1==m2)
        {
            accept(a,m1,n1);
            accept(b,m2,n2);
            mult(a,b,c,m1,n1,m2,n2);
            display(c,m1,n2);
        }
        else
        {
            printf("Multiplication impossible");
        }
    }
    return 0;

}
